packages/sls-env/src/__tests__/dependencies.test.ts

```diff
@@ -174,5 +174,89 @@ describe('serverless environment', () => {
         new EventDependencyError("No event based dependency found for 'buildMessage'")
       )
     })
+
+    it('correctly resolves dependency factories dependent on event', async () => {
+      type BuildMessage = (message: string, name: string) => string
+      type BuildMessageDependencies = {
+        buildMessage: DependencyFactory<BuildMessage>
+      }
+
+      const buildMessagesFactory = () => ({ event }: EventPayload) => {
```

Comment:
What is the return type for the function?

Reply:
It should be of type
```
type DependencyFactoryResolver<D, P = any, E = any, C = any> = (params: EventDependencyResolverParams<P, E, C>) => D
```